### PR TITLE
add wireguard example

### PR DESCRIPTION
**Pull request by 0oM4R**
Adding an explained example to deploy gateways with wireguard backends.
- #119

### CODE REVIEW

File: src/javascript/grid3_wireguard_gateway.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
+first, make sure you have your [client](./grid3_javascript_loadclient.md) prepared
```
Comment (xmonader) [inline src/javascript/grid3_wireguard_gateway.md:1]: replace with 

```
### client configurations

Check [client](./grid3_javascript_loadclient.md) to configure the client

```

File: src/javascript/grid3_wireguard_gateway.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+main();
+
+```
+
+### Detailed explanation
+
+What we need to do with that code is:
+
+- Deploy Vm with wireguard access.
+- Deploy gateway to access the machine through it.
```
Comment (xmonader) [inline src/javascript/grid3_wireguard_gateway.md:107]: not accurate, please change to  `Deploy a name contract with the wireguard IP as the backend`

File: src/javascript/grid3_wireguard_gateway.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  await grid3.disconnect();
+}
+
+main();
+
+```
+
+### Detailed explanation
+
+What we need to do with that code is: Deploy a name contract with the wireguard IP as the backend
```
Comment (AhmedHanafy725) [inline src/javascript/grid3_wireguard_gateway.md:107]: the `name contract` only contains the name, so maybe can be called `name gateway`

File: src/javascript/grid3_wireguard_gateway.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+// deploy the vm
+const vmResult = await grid3.machines.deploy(machines);
+log(vmResult);
+```
+
+- `CreateNetWorkModel` :
+ we are creating a network and set the node id to be `gwNode`, the name `monNetwork` and inside the function we set `addAccess: true` to add __wireguard__ access.
+
+- `createMachineModel` and `createMachinesModel` is similar to the previous section of [deploying a single VM](../javascript/grid3_javascript_vm.md), but we are passing the created `NetworkModel` to the machines model and the entry point here runs a py server .
```
Comment (AhmedHanafy725) [inline src/javascript/grid3_wireguard_gateway.md:163]: runs a simple python server

File: src/javascript/grid3_wireguard_gateway.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+  const deletedMachines = await grid3.machines.delete({ name:  machines.name});
+  log(deletedMachines);
+
+  const deletedGW = await grid3.gateway.delete_name({ name: gateway.name});
+  log(deletedGW);
+```
+
+### Summary
+
+This was a detailed description of how we can create a vm with private ip (wireguard) and use it as a backend for a gateway contract.
```
Comment (AhmedHanafy725) [inline src/javascript/grid3_wireguard_gateway.md:219]:
```suggestion
This was a detailed description of how we can create a vm with private ip (wireguard) and use it as a backend for a name gateway.
```

Comment (AhmedHanafy725) [reply src/javascript/grid3_wireguard_gateway.md:219]: or maybe `gateway workload`

**Review by AhmedHanafy725 — COMMENTED** [review]
can we also add the target of this deployment which is accessing a server inside the vm through the gateway using the private network (wireguard) as the backend?

and maybe also at the end add how to get the domain created to access this server

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**